# STAT 207 Lab 5 - Group Part - [5 total points] - Solutions

## Confidence Intervals

Due: Tuesday, September 28, 11:59pm CST

## <u>Case Study</u>: Former STAT107 Student Analysis

In [1]:
#imports here
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns; sns.set()

### Motivation
Starting next Fall, STAT107 and STAT207 (as well as other courses) will count as requirements towards a brand new suite of majors that we are calling "X+Data Science" majors. Thus, we in the statistics department are interested to learn more about the type of students that enroll in courses like STAT107. In the previous four semesters, 497 students enrolled in STAT107. Rather than track down and ask each of these 497 students questions, we will use the students in *this lab section* (who have presumably taken STAT107) in an attempt to learn more about the larger population of **all students who have taken STAT107 in the last four semesters**.

<hr>

### <u>Part 1</u>:  [1 pt] Create and Conduct a Survey

#### 1.1. Create a research question that *you* would be interested in asking all former STAT107 students (from the last four semesters).
   *  <u>Option 1</u>: Create a question that has a numerical response, for instance: "How many UIUC games have you attended so far?"
   * <u>Option 2</u>: Create a question that has a categorical response (with just two answers), for instance, "Are you team Jacob or team Edward?"

**<u>Note</u>: Don't use these example questions. You should come up with your own.**

**<u>Note</u>: If you were not able to attend the lab, then you can pretend that you asked the students in your lab this question: "Are you a sophomore?"**

Write your question below.

Are you a statistics major?

#### 1.2. Then, ask your question to 6 randomly selected students in this lab section and record your answers. Put these six responses in a dataframe.

*To ensure that your sample of students in this lab is random, see Yuhan or the course assistants for your random sample.*

**<u>Note</u>: If you were not able to attend the lab, then here is a random sample of six students from the class and their answers to your question (that I found out from the class roster page): ['no','no','no','no','yes','no']**
<hr>

In [17]:
df = pd.DataFrame({"response": ['no','no','no','no','yes','no']}, index = ['Ethan', 'Ding', 'Nicole', 'Zane', 'Junseok', 'Maulik'])

In [18]:
df

response
Ethan         no
Ding          no
Nicole        no
Zane          no
Junseok      yes
Maulik        no

<hr>

### <u> Part 2</u>: [1 pt] Calculate Sample Statistics

Then, using your sample that you just collected do one of the following.

**a. If your sample is comprised of numerical data:**

    * calculate the sample mean
    * calculate the sample standard deviation
    * plot the histogram of the numerical values in your sample
    
**b. If your sample is comprised of categorical data:**

    * calculate the proportion of observations in the sample that are of a particular value that you are interested in.

In [19]:
prop = len(df[df['response'] == 'yes']) / len(df)
prop

0.16666666666666666

<hr>

### <u>Part 3</u>: [2.5 pt] Create a Confidence Interval

Finally, using this sample we would like to create a plausible range of values for a **mean value** or **proportion value** corresponding to all students who took STAT107 in the last four semesters. Specifically, we will create a 90% confidence interval for this **population parameter**.
   *  For instance, if you asked six students in this lab "How many UIUC games have you attended so far?", then you should create a 90% confidence interval for the **average** number of games that **all former STAT107 students (from the last four semesters) have attended.**
   *  For instance, if you asked six students in this lab "Are you team Jacob or team Edward?", then you should create a 90% confidence interval for the **proportion** of **all former STAT107 students (from the last four semesters)** that are a **team Jacob.**
   
   
Make sure you check the appropriate corresponding Central Limit Theorem conditions needed in order to create this confidence inteval. *If you find that not all of these conditions are met, then you should "proceed with caution". This means that you should still create this confidence interval (for the purpose of this assignment), but knowing that the conclusions that we are drawing from this confidence interval may not be completely accurate.*

<hr>

In [15]:
from scipy.stats import norm
critical_value = norm.ppf(0.90)
critical_value

1.2815515655446004

In [20]:
lower_bound = prop - critical_value * (prop / np.sqrt(df.shape[0]))
upper_bound = prop + critical_value * (prop / np.sqrt(df.shape[0]))

In [21]:
print(lower_bound, ',', upper_bound)

0.07946812737085493 , 0.2538652059624784


### <u>Part 4</u>: [0.5 pt] Confidence Interval Interpretation

Put your confidence interval into words.

We are 90% confident that the proportion of students who are statistic majors out of all the students who took Stat107 in the past 4 semesters is between 0.079 and 0.254.

### <u>Contribution Report</u>:

Ethan Wee & Khushi Jain -- worked on all of the lab together.